cabal-install/Distribution/Solver/Modular/IndexConversion.hs

```diff
@@ -93,11 +94,18 @@ convIP :: SI.InstalledPackageIndex -> InstalledPackageInfo -> (PN, I, PInfo)
 convIP idx ipi =
   case mapM (convIPId (DependencyReason pn M.empty S.empty) comp idx) (IPI.depends ipi) of
         Nothing  -> (pn, i, PInfo [] M.empty M.empty (Just Broken))
-        Just fds -> ( pn
-                    , i
-                    , PInfo fds (M.singleton ExposedLib (IsBuildable True)) M.empty Nothing)
+        Just fds -> ( pn, i, PInfo fds components M.empty Nothing)
  where
+  -- TODO: Handle sub-libraries and visibility.
```

Comment:
What does this entail?

Reply:
I forgot to look into handling installed packages.  It could be tricky because of the hack where the solver treats installed internal libraries as packages with munged names:

https://github.com/haskell/cabal/blob/8f42d3fa9b1257f80b55cc9b35e9da1a2a8195f3/cabal-install/Distribution/Solver/Modular/IndexConversion.hs#L106-L131

I would need to figure out how to distinguish between inter- and intra-package dependencies when enforcing visibility.